cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -109,13 +121,127 @@ public Void call() throws Exception {
     }
   }
 
+  private void cleanFilesWithoutMeta(final long tillTime, final Map<String, NamespaceId> namespacedLogBaseDirMap,
+                                     final SetMultimap<String, Location> parentDirs) throws Exception {
+    LOG.info("Starting deletion of log files older than {} without metadata", tillTime);
+    List<NamespaceMeta> namespaceMetaList = namespaceQueryAdmin.list();
```

Comment:
no, why? we're doing impersonator.doAs, so the cleanup should succeed. Oh, unless the meta for the namespace itself is gone, so you don't know the impersonated user? Well in that case, we should probably document that users should remove the log files.